ui/src/Connection.elm

```diff
@@ -0,0 +1,224 @@
+    }
+
+
+initial : Model
+initial =
+    { response = ""
+    , things = Thing.initial
+    , channels = Channel.initial
+    , checkedThingsIds = []
+    , checkedChannelsIds = []
+    }
+
+
+type Msg
+    = Connect
+    | Disconnect
+    | ThingMsg Thing.Msg
+    | ChannelMsg Channel.Msg
+    | GotResponse (Result Http.Error String)
+    | CheckThing String
+    | CheckChannel String
+
+
+update : Msg -> Model -> String -> ( Model, Cmd Msg )
+update msg model token =
+    case msg of
+        Connect ->
+            if List.isEmpty model.checkedThingsIds || List.isEmpty model.checkedChannelsIds then
+                ( model, Cmd.none )
+
+            else
+                ( { model | checkedThingsIds = [], checkedChannelsIds = [] }
+                , Cmd.batch (connect model.checkedThingsIds model.checkedChannelsIds "PUT" token)
+                )
+
+        Disconnect ->
+            if List.isEmpty model.checkedThingsIds || List.isEmpty model.checkedChannelsIds then
+                ( model, Cmd.none )
+
+            else
+                ( { model | checkedThingsIds = [], checkedChannelsIds = [] }
+                , Cmd.batch (connect model.checkedThingsIds model.checkedChannelsIds "DELETE" token)
+                )
+
+        GotResponse result ->
+            case result of
+                Ok statusCode ->
+                    ( { model | response = statusCode }, Cmd.none )
+
+                Err error ->
+                    ( { model | response = Error.handle error }, Cmd.none )
+
+        ThingMsg subMsg ->
+            let
+                ( updatedThing, thingCmd ) =
+                    Thing.update subMsg model.things token
+            in
+            ( { model | things = updatedThing }, Cmd.map ThingMsg thingCmd )
+
+        ChannelMsg subMsg ->
+            let
+                ( updatedChannel, channelCmd ) =
+                    Channel.update subMsg model.channels token
+            in
+            ( { model | channels = updatedChannel }, Cmd.map ChannelMsg channelCmd )
+
+        CheckThing id ->
+            ( { model | checkedThingsIds = Helpers.checkEntity id model.checkedThingsIds }, Cmd.none )
+
+        CheckChannel id ->
+            ( { model | checkedChannelsIds = Helpers.checkEntity id model.checkedChannelsIds }, Cmd.none )
+
+
+
+-- VIEW
+
+
+view : Model -> Html Msg
+view model =
+    Grid.container []
+        [ Grid.row []
+            [ Grid.col []
+                [ Card.config
+                    []
+                    |> Card.headerH3 [] [ text "Things" ]
+                    |> Card.block []
+                        [ Block.custom
+                            (Table.table
+                                { options = [ Table.striped, Table.hover, Table.small ]
+                                , thead =
+                                    Table.simpleThead
+                                        [ Table.th [] [ text "Name" ]
+                                        , Table.th [] [ text "Id" ]
```

Comment:
Maybe better `ID`

Reply:
I am not sure that `ID` is idiomatic for Elm. It is for Go, but I think Elm uses `Id` and similar.

Reply:
I've the impression that's the title of the table. But maybe I'm wrong

Reply:
Yeah, that's a table's title. Let's leave it for master.